Objects/abstract.c

```diff
@@ -2881,7 +2881,47 @@ PyAIter_Check(PyObject *obj)
             tp->tp_as_async->am_anext != &_PyObject_NextNotImplemented);
 }
 
+static int
+iternext(PyObject *iter, PyObject **item)
+{
+    *item = (*Py_TYPE(iter)->tp_iternext)(iter);
+    if (*item == NULL) {
+        PyThreadState *tstate = _PyThreadState_GET();
+        /* When the iterator is exhausted it must return NULL;
+         * a StopIteration exception may or may not be set. */
+        if (!_PyErr_Occurred(tstate)) {
+            return 0;
+        }
+        if (_PyErr_ExceptionMatches(tstate, PyExc_StopIteration)) {
+            _PyErr_Clear(tstate);
+            return 0;
+        }
+        return -1;
+    }
+    return 1;
+}
+
+/* Return 1 and set 'item' to the next item of iter on success.
+ * Return 0 and set 'item' to NULL when there are no remaining values.
+ * Return -1, set 'item' to NULL and set an exception on error.
+ */
+int
+PyIter_NextItem(PyObject *iter, PyObject **item)
+{
+    assert(iter != NULL);
+    assert(item != NULL);
+
+    if (!PyIter_Check(iter) && !PyAIter_Check(iter)) {
+        PyErr_Format(PyExc_TypeError, "expected an iterator, not '%T'", iter);
```

Comment:
```suggestion
        PyErr_Format(PyExc_TypeError, "expected an iterator, got '%T'", iter);
```

I think we usually have the `expected ..., got` pattern but maybe I'm mistaken.

Reply:
Thanks, yes that's better!

Reply:
Usually I just write %T without the quotes.

Reply:
> Usually I just write %T without the quotes.

I believe it is better to include them. A large portion of our error messages include single quotes around the type name.